examples/main/main.cpp

```diff
@@ -36,6 +37,8 @@ void sigint_handler(int signo) {
         if (!is_interacting) {
             is_interacting=true;
         } else {
+            llama_print_timings(ctx);
+            llama_free(ctx);
```

Comment:
There's no harm in it, but I probably wouldn't bother to free `ctx` right before an `_exit` in an sigint handler. `_exit`'s purpose is to terminate the program immediately. Typically, you either clean up your memory properly and `exit`, or you don't and you use `_exit`.

Reply:
No worries, I wasn't aware of this. Its removed. :)